src/store.rs

```diff
@@ -2251,7 +1558,7 @@ impl<F: LurkField> Store<F> {
 
     fn hash_str(&self, str: Ptr<F>, mode: HashScalar) -> Option<ScalarPtr<F>> {
         if str.is_opaque() {
```

Comment:
all these `.is_opaque()` checks are redundant. We should merge the `hash_*` functions into one (based on ScalarExpression, per the tracking issue)